app/src/main/java/com/example/mytrainingpal/screens/TrainingsPreview.kt

```diff
@@ -0,0 +1,172 @@
+        val selectedExercises: MutableList<Pair<Exercise, ExerciseDetails>> = mutableListOf()
+
+        if (unsoreExercises.size < (setsNeeded / defaultSets)) {
+            // fill the workout first with unsore exercises
+            selectedExercises.addAll(unsoreExercises.map { Pair(it, defaultDetails) })
+            setsLeftover -= (selectedExercises.size * defaultSets)
+            // then fill the workout with slightly sore exercises of full default sets
+            if (setsLeftover > defaultSets) {
+                val slightlySoreExercisesNeeded = setsLeftover / defaultSets
+                selectedExercises.addAll(
+                    slightlySoreExercises.asSequence().shuffled().take(slightlySoreExercisesNeeded)
+                        .map { Pair(it, defaultDetails) })
+                setsLeftover -= defaultSets * slightlySoreExercisesNeeded
+            }
+        } else {
+            selectedExercises.addAll(
+                unsoreExercises.asSequence().shuffled().take(setsNeeded / defaultSets)
+                    .map { Pair(it, defaultDetails) })
+            setsLeftover -= (selectedExercises.size * defaultSets)
+        }
+        // then increase the amount of sets one after another till the time is up
+        var i = 0
+        while (setsLeftover > 0) {
+            selectedExercises[i] = Pair(
+                selectedExercises[i].first, ExerciseDetails(
+                    sets = selectedExercises[i].second.sets + 1,
+                    reps = selectedExercises[i].second.reps,
+                    weight = selectedExercises[i].second.weight
+                )
+            )
+            setsLeftover--
+            i++
+            if (i >= selectedExercises.size) {
+                i = 0
+            }
+        }
+        exercises.clear()
+        exercises.addAll(selectedExercises)
+
+    }
+
+    TabScreen(
+        tabContent = {
+            TrainingsPreviewScreenContent(
+                exercises = exercises,
+                duration
+            )
+        },
+        topBarTitle = Screen.TrainingsPreview.label,
+        topBarIcon = Screen.TrainingsPreview.icon,
+        navController = navController,
+        floatingActionButton = {
+            StartExerciseFloatingButton {
+                navController.navigate(Screen.Home.route)
+            }
+        }
+    )
+}
+
+@Composable
+fun TrainingsPreviewScreenContent(
+    exercises: MutableList<Pair<Exercise, ExerciseDetails>>,
+    duration: IntHolder
+) {
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = Modifier
+            .fillMaxHeight()
+            .verticalScroll(rememberScrollState()),
+    ) {
+        //TextIconWidget(text = "Warm up", imageVector = Icons.Default.LocalFireDepartment)
+        TextIconWidget(
+            text = "Your ${duration.value} min Routine",
+            imageVector = Icons.Default.FitnessCenter
+        )
+        exercises.forEach() { (exercise, details) ->
```

Comment:
Please use a lazy list (`LazyColumn`) here instead of always rendering all items. 
Here is the docs link for reference: https://developer.android.com/jetpack/compose/lists

Reply:
We do not really need a lazy column, since there are not that many elements, which makes the performance factor negligible.